src/components/MoneyRequestConfirmationList.tsx

```diff
@@ -520,7 +521,9 @@ function MoneyRequestConfirmationList({
         if (updatedTagsString !== TransactionUtils.getTag(transaction) && updatedTagsString) {
             IOU.setMoneyRequestTag(transactionID, updatedTagsString);
         }
-    }, [policyTagLists, transaction, transactionID, policyTags, canUseViolations]);
+        // Keep 'transaction' out to ensure that we autoselect the option only once
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [policyTagLists, policyTags, canUseViolations]);
```

Comment:
We should have kept the `transactionID` here because at the first render there is no transaction and on the second render this effect does not run which caused https://github.com/Expensify/App/issues/52879. Also the `transactionID` does not change afterwards i.e. does not cause problems as the `transaction`.